pkg/security/agent/agent.go

```diff
@@ -0,0 +1,113 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+package agent
+
+import (
+	"context"
+	"errors"
+	"io"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/logs/message"
+	"github.com/DataDog/datadog-agent/pkg/util"
+	"google.golang.org/grpc"
+
+	coreconfig "github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/logs/config"
+	"github.com/DataDog/datadog-agent/pkg/security/api"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// RuntimeSecurityAgent represents the main wrapper for the Runtime Security product
+type RuntimeSecurityAgent struct {
+	logClient *DDClient
+	conn      *grpc.ClientConn
+	running   atomic.Value
+	wg        *sync.WaitGroup
+}
+
+// NewRuntimeSecurityAgent instantiates a new RuntimeSecurityAgent
+func NewRuntimeSecurityAgent() (*RuntimeSecurityAgent, error) {
+	hostname, err := util.GetHostname()
+	if err != nil {
+		return nil, err
+	}
+
+	socketPath := coreconfig.Datadog.GetString("runtime_security_config.socket")
+	if socketPath == "" {
+		return nil, errors.New("runtime_security_config.socket must be set")
+	}
+
+	path := "unix://" + socketPath
+	conn, err := grpc.Dial(path, grpc.WithInsecure())
```

Comment:
Could you please add a reasonable retrier here and possibly increase the timeout option here (would be nice to add a test for that too). There is a race right now - it fail to dial if the sysprobe container didn't bring up the socket yet (running in separate containers). To confirm the socket actually exists:
```
# inside the security-agent container:
root@datadog-monitoring-8nvff:/# ls /var/run/sysprobe/
runtime-security.sock  sysprobe.sock
```

Reply:
There already is a retry mechanism so that we don't fail if system-probe hasn't created the socket yet (agent.go:88)

Reply:
Hm.. maybe it actually does connect after retries, just there isn't an indication in the logs it succeeded eventually. Maybe add a log saying "successfully connected" to socket?

Reply:
I added a log when there is a connection success after each connection fail